src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -546,9 +547,16 @@ where
 
                 let mut timer = Instant::now();
 
-                // We just use an advancing number for our capability. No one cares
-                // about what this actually is, downstream.
-                let mut batch_counter = timely::progress::Timestamp::minimum();
+                // The timestamp that we use for emitting untimestamped batches.
+                // This doesn't _have_ to be wall-clock time, but it's easy
+                // enough to have these somewhat synchronized between all the
+                // workers, which ensures that downstream operators can always
+                // make progress.
+                //
+                // NOTE: This is separate from the wall-clock time that we use
+                // to timestamp messages. Though in practice they're the same,
+                // right now.
+                let mut emit_ts = Timestamp::from((*now_fn)());
```

Comment:
`NowFn` is not guaranteed to be strictly monotonic, we should add some handling logic here

Reply:
I'm adding a `std::cmp::max()` around the call where I update it? Or do you want more for now?